### PR TITLE
Fix the hasJoinOperation when SQL statement has limit

### PR DESCRIPTION
**Pull request by yukuai518**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-215

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
  Add instanceOf check when parseQuery returns a SqlNode type

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
    Add some test cases where SQL statement has a limit keyword

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java [hunk 1]
```diff
@@ -1148,9 +1149,19 @@ public static boolean hasJoinOperation(String selectQuery) {
 
     SqlParser sqlParser = SqlParser.create(selectQuery);
     try {
-      SqlSelect sqlSelect = (SqlSelect)sqlParser.parseQuery();
-      SqlNode node = sqlSelect.getFrom();
-      return node.getKind() == SqlKind.JOIN;
+
+      SqlNode all = sqlParser.parseQuery();
+      SqlSelect query;
+      SqlNode from;
```
Comment (zxcware) [inline gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java]: With `LIMIT` in the query, the previous approach failed?

**Comment by abti** [comment]
Please create an associated Apache Jira and link it here.

File: gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java [hunk 2]
```diff
@@ -1148,9 +1149,19 @@ public static boolean hasJoinOperation(String selectQuery) {
...
+      if (all instanceof SqlSelect) {
+        query = (SqlSelect) all;
+      } else if (all instanceof SqlOrderBy) {
+        query = (SqlSelect) ((SqlOrderBy) all).query;
+      } else {
+        throw new UnsupportedOperationException("The select query is type of " + all.getClass() + " which is not supported here");
+      }
+      from = query.getFrom();
+      return from.getKind() == SqlKind.JOIN;
```
Comment (zxcware) [inline gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java]: minor: Can remove the declaration of `SqlNode from`. Just `return query.getFrom().getKind() == SqlKind.JOIN`

File: gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java (hunk 1 above)
Comment (yukuai518) [inline gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java]: @zxcware Yes it is

File: gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java (hunk 2 above)
Comment (yukuai518) [inline gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java]: @zxcware Fixed.

**Comment by yukuai518** [comment]
@abti Jira ticket is created.

**Pull request closed without merging**